release/models/network-instance/openconfig-network-instance-l2.yang

```diff
@@ -376,4 +383,308 @@ submodule openconfig-network-instance-l2 {
+                  uses l2ni-l2rib-mac-table-producer-state;
+                }
+              }
+            }
+
+            uses l2ni-l2rib-common-next-hop-state;
+          }
+        }
+
+      }
+
+      container mac-ip-table {
+        description
+          "Oper state container for MAC-IP address information installed into
+           the MAC VRF of the L2RIB";
+
+        container entries {
+          description
+            "Enclosing container for list of MAC-IP address entries";
+
+          list entry {
+            key "mac-address vlan host-ip";
+            description "List of learned MAC-IP addresses";
+
+            leaf mac-address {
+              type leafref {
+                path "../state/mac-address";
+              }
+              description "Leafref of MAC-IP address object";
+            }
+
+            leaf vlan {
+              type leafref {
+                path "../state/vlan";
+              }
+              description
+                "Leafref VLAN on which the MAC-IP address is present";
+            }
+
+            leaf host-ip {
+              type leafref {
+                path "../state/host-ip";
+              }
+                description "Leafref host-ip for the MAC-IP table entry";
+            }
+
+            container state {
+              config false;
+              description
+                "Operational state data for L2RIB MAC-IP table object entry";
+              uses l2ni-l2rib-mac-ip-table-state;
+            }
+
+            container producers {
+              description "Source producers for each MAC-IP Table entry";
+              list producer {
+                key "producer";
+                description
+                  "List of producers for each MAC-IP table entry";
+
+                leaf producer {
+                  type leafref {
+                    path "../state/producer";
+                  }
+                  description
+                    "Reference to producer list key";
+                }
+
+                container state {
+                  config false;
+                  description "State container for L2RIB MAC Table Entries";
+                  uses l2ni-l2rib-common-producer-state;
+                }
+              }
+            }
+
+            uses l2ni-l2rib-common-next-hop-state;
+          }
+        }
+      }
+    }
+  }
+
+  grouping l2ni-l2rib-mac-table-state {
+    description "L2RIB MAC Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+  }
+
+  grouping l2ni-l2rib-mac-ip-table-state {
+    description "L2RIB Mac-IP Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+
+    leaf host-ip {
+      type inet:ip-address;
+      description "Host IP address for the L2RIB MAC-IP entry";
+    }
+
+    leaf l3-vni {
+      type oc-evpn-types:evi-id;
+      description "Layer3 VNI used for inter-subnet routing";
+    }
+  }
+
+  grouping l2ni-l2rib-common-state {
```

Comment:
Please use the grouping naming convention:

Reply:
This is ok now as discussed.